tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/e2e-upgrade-test.go

```diff
@@ -102,15 +66,58 @@ var _ = Describe(
 				_, err = newIbguBuilder.WaitUntilComplete(30 * time.Minute)
 				Expect(err).NotTo(HaveOccurred(), "Prep and Upgrade IBGU did not complete in time.")
 
+				upgradeSuccessful = true
 			})
 
 			By("Saving target sno cluster info after upgrade", func() {
 				err := cnfclusterinfo.PostUpgradeClusterInfo.SaveClusterInfo()
 				Expect(err).ToNot(HaveOccurred(), "Failed to collect and save target sno cluster info after upgrade")
 			})
 
+			By("Running post-upgrade validations", func() {
+				cnfibuvalidations.PostUpgradeValidations()
+			})
 		})
 
-		cnfibuvalidations.PostUpgradeValidations()
+		if upgradeSuccessful {
+			It("Rollback successful upgrade", reportxml.ID("69058"), func() {
+				By("Deleting upgrade ibgu created on target hub cluster", func() {
+					newIbguBuilder := ibgu.NewIbguBuilder(cnfinittools.TargetHubAPIClient,
+						tsparams.IbguName, tsparams.IbguNamespace).
+						WithClusterLabelSelectors(tsparams.ClusterLabelSelector).
+						WithSeedImageRef(cnfinittools.CNFConfig.IbguSeedImage, cnfinittools.CNFConfig.IbguSeedImageVersion).
+						WithOadpContent(cnfinittools.CNFConfig.IbguOadpCmName, cnfinittools.CNFConfig.IbguOadpCmNamespace).
+						WithPlan([]string{"Prep", "Upgrade"}, 5, 30)
+
+					_, err := newIbguBuilder.DeleteAndWait(1 * time.Minute)
+					Expect(err).ToNot(HaveOccurred(), "Failed to delete prep-upgrade ibgu on target hub cluster")
+
+					rollbackIbguBuilder := ibgu.NewIbguBuilder(cnfinittools.TargetHubAPIClient, "rollbackibgu", tsparams.IbguNamespace)
+					rollbackIbguBuilder = rollbackIbguBuilder.WithClusterLabelSelectors(tsparams.ClusterLabelSelector)
+					rollbackIbguBuilder = rollbackIbguBuilder.WithSeedImageRef(
+						cnfinittools.CNFConfig.IbguSeedImage,
+						cnfinittools.CNFConfig.IbguSeedImageVersion)
+					rollbackIbguBuilder = rollbackIbguBuilder.WithOadpContent(
+						cnfinittools.CNFConfig.IbguOadpCmName,
+						cnfinittools.CNFConfig.IbguOadpCmNamespace)
+					rollbackIbguBuilder = rollbackIbguBuilder.WithPlan([]string{"Rollback", "FinalizeRollback"}, 5, 30)
+
+					rollbackIbguBuilder, err = rollbackIbguBuilder.Create()
+					Expect(err).ToNot(HaveOccurred(), "Failed to create rollback Ibgu.")
+
+					_, err = rollbackIbguBuilder.WaitUntilComplete(30 * time.Minute)
+					Expect(err).NotTo(HaveOccurred(), "Rollback IBGU did not complete in time.")
+
+					_, err = rollbackIbguBuilder.DeleteAndWait(1 * time.Minute)
```

Comment:
Just curious - why are we deleting the rollback ibgu at the end? 
Also there should be a step logging?

Reply:
@yliu127 We delete the rollback ibgu at the end of the test so that the target env is in the same state it was in prior to the test. This way, if we need to re-run the test suite we can do so without having to manually remove the rollback IBGU.